src/components/select/multi-select/multi-select.test.js

```diff
@@ -0,0 +1,1318 @@
+        "have.css",
+        "background-color",
+        "rgba(0, 0, 0, 0)"
+      );
+    });
+
+    it("should check table header content in list with multiple columns", () => {
+      CypressMountWithProviders(<MultiSelectMultiColumnsComponent />);
+
+      const headerCol1 = "Name";
+      const headerCol2 = "Surname";
+      const headerCol3 = "Occupation";
+
+      dropdownButton().click();
+      multiColumnsSelectListHeaderColumn(1)
+        .should("have.text", headerCol1)
+        .and("be.visible");
+      multiColumnsSelectListHeaderColumn(2)
+        .should("have.text", headerCol2)
+        .and("be.visible");
+      multiColumnsSelectListHeaderColumn(3)
+        .should("have.text", headerCol3)
+        .and("be.visible");
+    });
+
+    it("should indicate a matched filtered string with bold and underline", () => {
+      CypressMountWithProviders(<MultiSelectMultiColumnsComponent />);
+
+      const text = "Do";
+
+      commonDataElementInputPreview().click().should("be.focused");
+      commonDataElementInputPreview().type(text);
+      boldedAndUnderlinedValue(text)
+        .should("have.css", "text-decoration-line", "underline")
+        .and("have.css", "text-decoration-style", "solid")
+        .and("have.css", "font-weight", "700");
+    });
+
+    it.each([["Xyz"], [" "]])(
+      'should indicate no results match entered string "%s"',
+      (text) => {
+        CypressMountWithProviders(<MultiSelectMultiColumnsComponent />);
+
+        commonDataElementInputPreview().click().should("be.focused");
+        commonDataElementInputPreview().type(text);
+        selectList().should("be.visible");
+        multiColumnsSelectListHeader()
+          .should("have.length", columns)
+          .and("be.visible");
+        multiColumnsSelectListNoResultsMessage(text).should("be.visible");
+      }
+    );
+
+    it.each([
+      ["3", "Blue"],
+      ["7", "Pink"],
+    ])(
+      "should set defaultValue prop to %s and show option pill %s preselected",
+      (value, option) => {
+        CypressMountWithProviders(
+          <MultiSelectDefaultValueComponent defaultValue={[value]} />
+        );
+
+        multiSelectPill().should("have.attr", "title", option);
+      }
+    );
+
+    it("should have no pill option preselected if defaultValue prop is not set", () => {
+      CypressMountWithProviders(<MultiSelectDefaultValueComponent />);
+
+      multiSelectPill().should("not.exist");
+    });
+
+    it("should render Multi Select with custom coloured pills", () => {
+      CypressMountWithProviders(<MultiSelectCustomColorComponent />);
+
+      for (let i = 0; i < 2; i++) {
+        let borderColor, pillColor;
+        if (i === 0) {
+          borderColor = "rgb(255, 191, 0)";
+          pillColor = borderColor;
+        }
+        if (i === 1) {
+          borderColor = "rgb(0, 119, 200)";
+          pillColor = "rgba(0, 0, 0, 0)";
+        }
+
+        multiSelectPillByPosition(i)
```

Comment:
it is much easier to put those 2 values here - instead of wrapping it in a `for` loop and `if` statement

Reply:
I just love a good `for` loop and `if` statement :D